src/vs/base/common/lifecycle.ts

```diff
@@ -244,10 +309,23 @@ export class MutableDisposable<T extends IDisposable> implements IDisposable {
 
 	dispose(): void {
 		this._isDisposed = true;
-		markTracked(this);
+		markAsDisposed(this);
 		this._value?.dispose();
 		this._value = undefined;
 	}
+
+	/**
+	 * Clears the value, but does not dispose it.
+	 * The old value is returned.
+	*/
+	clearAndLeak(): T | undefined {
```

Comment:
I could not find a usage of this? Is this needed?

Reply:
This is needed in inline completions. Currently they use a copy of `MutableDisposable` that has this method, as I didn't want to be blocked by a code review.

Also, `clearAndLeak` makes use of the new tracking API (disposables can be removed from parent disposables).